crates/core/tedge_mapper/src/core/mapper.rs

```diff
@@ -121,12 +121,45 @@ impl Mapper {
             self.process_message(message).await;
         }
 
-        // Continue processing messages after the sync period
-        while let Some(message) = self.input.next().await {
-            self.process_message(message).await;
+        // Create inotify steam for capturing the inotify events.
+        let mut inotify = create_inotify_watch(ops_dir.clone()).expect("No operations directory");
+        let mut buffer = [0; 1024];
+        let mut stream = inotify
+            .event_stream(&mut buffer)
+            .expect("Failed to create the inotify stream");
+
+        loop {
+            tokio::select! {
+                msg =  self.input.next() => {
+                            match msg {
+                                Some(message)=> {self.process_message(message).await;}
+                                None => {error!("Failed to extract the mqtt message");}
```

Comment:
Receiving no message is not an error. There is simply no more message to process and the loop is done.